experimental/framework/frs-client/src/FrsSimpleUrlResolver.ts

```diff
@@ -0,0 +1,42 @@
+import { IRequest } from "@fluidframework/core-interfaces";
+import { IUrlResolver, IFluidResolvedUrl, IResolvedUrl } from "@fluidframework/driver-definitions";
+import { IUser, ScopeType } from "@fluidframework/protocol-definitions";
+import { generateToken } from "@fluidframework/server-services-client";
+import { FrsConnectionConfig } from "./interfaces";
+
+export class FrsSimpleUrlResolver implements IUrlResolver {
+    constructor(
+        private readonly config: FrsConnectionConfig,
+        private readonly documentId: string,
+        private readonly user?: IUser,
+    ) { }
+
+    public async resolve(request: IRequest): Promise<IFluidResolvedUrl> {
+        const containerId = request.url.split("/")[0];
+        const token = generateToken(
+            this.config.tenantId,
+            this.documentId,
+            this.config.key,
```

Comment:
If a TokenProvider is included in the configuration, I'm thinking we should use that whenever possible for retrieving a token. That way this has the capability of being "secure." In this current state, a developer is always required to enter their tenant key in the code, so it is inherently insecure.

In short, if TokenProvider is provided, consumers should not be also required to provide a tenant key.

Reply:
So for the `ITokenProvider` interface, I was seeing calls for fetching the orderer/storage tokens. How do we use this to fetch the tenant key? Or do we just directly use one of the two tokens here and, if so, which one?

Reply:
You're only using the key to generate a token, right? So if you have a token provider, you should be able to just fetch the token rather than generate one yourself. For FRS purposes, Storage and Orderer tokens are interchangeable, so just pick one of those methods. Probably Orderer would be best.

Reply:
Got it, appreciate the explanation @znewton! I've updated it such that the `FrsConnectionConfig` takes in either a `key` or a `tokenProvider`. If no `tokenProvider` is supplied, the `FrsClient` creates an `InsecureTokenProvider`.
The `FrsUrlResolver` now just takes in the token provider from the `FrsClient`, either the newly created one or the one supplied from the config